src/Maestro/maestro-angular/src/app/page/channel-graph/channel-graph.component.ts

```diff
@@ -0,0 +1,136 @@
+import { Component, Input, OnChanges, SimpleChanges, AfterContentInit } from '@angular/core';
+import { graphlib, render } from 'dagre-d3';
+import { select } from 'd3';
+
+import { FlowGraph, FlowRef, FlowEdge } from 'src/maestro-client/models';
+import { RepoNamePipe } from 'src/app/pipes/repo-name.pipe';
+import { analyzeAndValidateNgModules } from '@angular/compiler';
+
+function getRepositoryShortName(repo:string): string {
+  return `${RepoNamePipe.prototype.transform(repo)}`;
```

Comment:
Why quote this into a string? why not just `return RepoNamePipe.prototype.transform(repo)`?

Reply:
The transform function returns a string | undefined, which is why I put it into a string. I can cast it, if that's better, or I can have the function return string | undefined, but I'll have to be doing some casting somewhere.

Reply:
Oooo. That's super not clear from this that there is implicit undefined flattening happening. In that case, I'd make it `return RepoNamePipe.prototype.transform(repo) || 'some default string'`